src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs

```diff
@@ -638,7 +638,7 @@ private void WriteFields(TypeBuilderImpl typeBuilder, BlobBuilder fieldDataBuild
                 Debug.Assert(field._handle == handle);
                 WriteCustomAttributes(field._customAttributes, handle);
 
-                if (field._offset > 0 && (typeBuilder.Attributes & TypeAttributes.ExplicitLayout) != 0)
+                if (field._offset >= 0 && (typeBuilder.Attributes & TypeAttributes.ExplicitLayout) != 0)
```

Comment:
The reason it was checking greater than 0 was because the default is 0, could you set the default to -1?
 
https://github.com/dotnet/runtime/blob/e0f6b9f91c92f0b76ab032fc2fdaeded52cb736a/src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/FieldBuilderImpl.cs#L23

Reply:
The constructor already does this.

https://github.com/dotnet/runtime/blob/e0f6b9f91c92f0b76ab032fc2fdaeded52cb736a/src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/FieldBuilderImpl.cs#L35

Reply:
Oh, right, thanks!